Sources/SwiftKafka/KafkaProducer.swift

```diff
@@ -193,8 +183,15 @@ public final class KafkaProducer: Service, Sendable {
         while !Task.isCancelled {
             let nextAction = self.stateMachine.withLockedValue { $0.nextPollLoopAction() }
             switch nextAction {
-            case .poll(let client):
-                client.poll(timeout: 0)
+            case .poll(let client, let source):
+                let events = client.eventPoll()
```

Comment:
Maybe I am missing something but in the case where our async sequence is shutdown we still have to poll right? Otherwise kafka will just queue up the events.

Reply:
Yes, this is implemented in another PR (#84) though 😄